samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/ImageryTileLayers.razor

```diff
@@ -0,0 +1,163 @@
+        <option value="@StretchType.StandardDeviation">standard deviation</option>
+    </select>
+    <br />
+    <label id="sliderLabel" class="esri-feature-form__label">
+        @_stretchType.ToString().ToKebabCase() value:
+    </label>
+    <div id="value-slider" class="slider"></div>
+    <input type="checkbox" checked @onchange="ToggleBlendMode" />
+    <label>Toggle blendMode</label>
+</div>
+
+
+@code {
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (_layer!.Renderer == null)
+        {
+            await _layer.Load();
+            _layer.Renderer = new RasterStretchRenderer(new AlgorithmicColorRamp(Algorithm.CieLab,
+                new MapColor(0, 0, 0, 0), new MapColor(0, 0, 0, 1)),
+                stretchType: _stretchType, 
+                statistics: [new RasterStatistics(120, 3068, 400, 700)]);
+            _layer.LayerChanged = true;
+            _rasterInfo = await _layer.GetServiceRasterInfo();
+            StateHasChanged();
+        }
+    }
+
+    private async Task OnSliderChanged(double[] values)
+    {
+        await UpdateRenderer(values);
+    }
+
+    private async Task UpdateRenderer(double[] values)
+    {
+        var bandStat = _rasterInfo!.Statistics[0];
+        RasterStretchRenderer newRenderer;
+        switch (_stretchType)
+        {
+            case StretchType.MinMax:
+                newRenderer = new RasterStretchRenderer(new AlgorithmicColorRamp(Algorithm.CieLab,
+                        new MapColor(0, 0, 0, 0), new MapColor(0, 0, 0, 1)),
+                    stretchType: _stretchType, 
+                    statistics: [bandStat with { Min = values[0], Max = values[1] }]);
+
+                break;
+            default:
+                newRenderer = new RasterStretchRenderer(new AlgorithmicColorRamp(Algorithm.CieLab,
+                        new MapColor(0, 0, 0, 0), new MapColor(0, 0, 0, 1)),
+                    stretchType: _stretchType, 
+                    statistics: [bandStat],
+                numberOfStandardDeviations: (int)values[0]);
+
+                break;
+        }
+        
+        await _layer!.SetRenderer(newRenderer);
+    }
+
+    private async Task ToggleBlendMode(ChangeEventArgs e)
+    {
+        if (e.Value is true)
+        {
+            await _layer!.SetProperty(nameof(ImageryTileLayer.BlendMode), BlendMode.DestinationIn);
+        }
+        else
+        {
+            await _layer!.SetProperty(nameof(ImageryTileLayer.BlendMode), BlendMode.Normal);
+        }
+    }
+    
+    private async Task ToggleStretchType(ChangeEventArgs e)
+    {
+        _stretchType = (StretchType)Enum.Parse(typeof(StretchType), e.Value?.ToString()!);
+
+        switch (_stretchType)
+        {
+            case StretchType.MinMax:
+                await _slider!.SetMax(5000);
+                await _slider.SetMin(0);
+                await _slider.SetStepInterval(10);
+                await _slider.SetValues([120, 3068]);
```

Comment:
do these async methods need to finish in order? I was wondering if there was something that could be implemented that's similar to Promise.all() in JS, where an array of async promises can be fired at once and when all of them finish, continue on.

Reply:
C# does have that feature, something like below, but it's wordy and I'm not sure if it's worth doing in this scenario.

```csharp
List<Task> tasks = [];
tasks.Add(Task.Run(() => _slider.SetMax(5000)));
tasks.Add(Task.Run(() => ...:
...
...

await Task.WhenAll(tasks);